apps/nextjs-website/src/components/organisms/GitBookContent/components/Swagger/Swagger.tsx

```diff
@@ -1,38 +1,107 @@
-import { Card, Typography } from '@mui/material';
+`use client`;
+import Accordion from '@mui/material/Accordion';
+import MuiAccordionSummary, {
+  AccordionSummaryProps,
+} from '@mui/material/AccordionSummary';
+import AccordionDetails from '@mui/material/AccordionDetails';
+import Typography from '@mui/material/Typography';
 import { SwaggerProps } from 'gitbook-docs/markdoc/schema/swagger';
+
+import { ReactNode } from 'react';
+import { useSpec } from './hooks/useSpec';
 import { useTranslations } from 'next-intl';
+import { Operations } from './Operations';
 import { OpenAPIV3 } from 'openapi-types';
+import { styled } from '@mui/material';
+import ArrowForwardIosSharpIcon from '@mui/icons-material/ArrowForwardIosSharp';
 
-import { Operations } from './Operations';
-import { useSpec } from './hooks/useSpec';
+const AccordionSummary = styled((props: AccordionSummaryProps) => (
+  <MuiAccordionSummary
+    expandIcon={<ArrowForwardIosSharpIcon sx={{ fontSize: '0.75rem' }} />}
+    {...props}
+  />
+))(() => ({
+  flexDirection: 'row-reverse',
+  '& .MuiAccordionSummary-expandIconWrapper.Mui-expanded': {
+    transform: 'rotate(90deg)',
+  },
+}));
 
-const Swagger = ({ path, method, src }: SwaggerProps) => {
-  const { spec, loading } = useSpec(src);
+const Swagger = ({
+  method,
+  summary,
+  children,
+  src,
+  path,
+}: SwaggerProps<ReactNode>) => {
+  const { spec } = useSpec(src);
   const t = useTranslations('swagger');
 
-  if (loading) return null;
-  if (!spec || !src)
+  if (spec && src) {
+    const validOperations = { [path]: [method] as OpenAPIV3.HttpMethods[] };
     return (
-      <Card sx={{ borderRadius: 1, p: 2, mb: 2 }} variant='outlined'>
-        <Typography sx={{ fontWeight: 'bold' }} variant='body1'>
-          {t('errorCard.header')}
-        </Typography>
-        <Typography
-          variant='body2'
-          sx={{ color: (theme) => theme.palette.text.secondary }}
-        >
-          {t('errorCard.message')}
-        </Typography>
-      </Card>
+      <Operations
+        spec={spec as OpenAPIV3.Document}
+        validOperations={validOperations}
+      />
     );
-
-  const validOperations = { [path]: [method] as OpenAPIV3.HttpMethods[] };
+  }
 
   return (
-    <Operations
-      spec={spec as OpenAPIV3.Document}
-      validOperations={validOperations}
-    />
+    <Accordion
+      sx={{
+        border: '1px solid #ddd',
+        borderRadius: '4px',
+        marginBottom: '16px',
+      }}
+    >
+      <AccordionSummary aria-controls='panel2-content' id='panel2-header'>
+        <div
+          style={{
+            padding: '8px 12px',
+            display: 'flex',
+            flexDirection: 'column',
+            alignItems: 'flex-start',
+          }}
+        >
+          <Typography
+            sx={{
+              backgroundColor: '#008847',
```

Comment:
We should avoid hardcoded colors (see also line 53) and extend the MUI theme by [adding a custom color](https://mui.com/material-ui/customization/palette/#custom-colors)